pygmt/clib/session.py

```diff
@@ -618,11 +626,31 @@ def call_module(self, module, args):
             argtypes=[ctp.c_void_p, ctp.c_char_p, ctp.c_int, ctp.c_void_p],
             restype=ctp.c_int,
         )
+        if isinstance(args, str):
+            # 'args' is a single text string with multiple options.
+            # E.g., "-R0/1/0/1 -JX10c -BWSen+tMy Title -Bxaf -Byaf".
+            # This is the way we used to pass arguments in PyGMT v0.11.0 and before.
+            mode = self["GMT_MODULE_CMD"]
+            argv = args.encode()
+        elif isinstance(args, list):
+            # 'args' is a list of strings and each string is an option.
+            # E.g., ["-R0/1/0/1", "-JX10c", "-BWSen+tMy Title", "-Bxaf", "-Byaf"].
+            # In this way, GMT knows that each list element is an option with arguments,
+            # so it won't incorrectly parse strings with whitespaces or quotation marks.
+            # This is the preferred way to pass arguments to GMT API, because we don't
+            # have to deal with whitespaces or quotation marks on the PyGMT side.
+            mode = len(args)  # Now mode is the number of arguments.
+            if mode == 0:  # No arguments, passing a null pointer
+                argv = None
+            else:
+                argv = (ctp.c_char_p * len(args))()
+                argv[:] = np.char.encode(args)
```

Comment:
Refactor after PR #3137 is merged.
```suggestion
                argv = strings_to_ctypes_array(args)
```